luminoth/datasets/object_detection_dataset.py

```diff
@@ -90,19 +91,36 @@ def _augment(self, image, bboxes, default_prob=0.5):
             aug_fn = DATA_AUGMENTATION_STRATEGIES[aug_type]
 
             random_number = tf.random_uniform([])
-            prob = aug_config.pop('prob', default_prob)
+            prob = tf.to_float(aug_config.pop('prob', default_prob))
             apply_aug_strategy = tf.less(random_number, prob)
 
             augmented = tf.cond(
                 apply_aug_strategy,
-                lambda: aug_fn(image, bboxes, **aug_config),
+                lambda: aug_fn(image, aug_config, bboxes),
                 lambda: {'image': image, 'bboxes': bboxes}
             )
 
-            applied_data_augmentation.append({aug_type: apply_aug_strategy})
+            update_condition = tf.greater(
```

Comment:
It makes more sense to have the "update condition" inside the data augmentation function, especially since flip and possible others strategies are always valid.

Reply:
Agreed.